singer-commons/src/main/java/com/pinterest/singer/loggingaudit/client/AuditEventKafkaSender.java

```diff
@@ -232,13 +255,13 @@ public void run() {
     while (!cancelled.get()) {
       try {
         refreshPartitionIfNeeded();
+        resetDefaultPartitionIfNeeded();
```

Comment:
Shouldn't resetDefaultPartition be called within refreshPartition? Partition Id setting can be done in one method instead of two.
Also setting valid partitionId should only be done during refresh interval checks for efficiency reasons else `resetDefaultPartitionIfNeeded` method will unnecessarily run the same logic again.